charts/egressd/templates/exporter/deployment.yaml

```diff
@@ -49,17 +49,20 @@ spec:
         {{- end }}
         {{- end }}
         env:
-            {{- if .Values.castai.apiKey }}
-        - name: CASTAI_API_KEY
-          valueFrom:
-            secretKeyRef:
-              name: "{{- include "egressd.fullname" . }}"
-              key: API_KEY
-            {{- end }}
         - name: POD_NAMESPACE
           valueFrom:
             fieldRef:
               fieldPath: metadata.namespace
+        envFrom:
+          - secretRef:
```

Comment:
Are we sure we want to use `envFrom`?

I see such drawbacks:
- chart users can't specify a different secret key, i.e. secret key `API_KEY` is hardcoded
- if we provide a secret which is missing `API_KEY`, the service will fail with a cryptic-ish error. If we used `env`, k8s will fail to start the pod with a proper error indicating that the key is missing.
- `secretKeyRef` is misleading here: it is usually expected to be `{name: $secretName, key: $secretKey}`, but we use string.

Reply:
I propose to use env:
```yaml
       {{- if or .Values.castai.apiKey .Values.castai.apiKeySecretKeyRef }}
        - name: CASTAI_API_KEY
          valueFrom:
            secretKeyRef:
              {{- if .Values.castai.apiKey }} {{/* apiKey takes precedence over secretKeyRef */}}
              name: "{{- include "egressd.fullname" . }}"
              key: API_KEY
              {{- else if  .Values.castai.apiKeySecretKeyRef }}
              {{toYaml .Values.castai.apiKeySecretKeyRef  | nindent 10}} {{/* 10 is prolly the wrong amount */}}
              {{- end }}
        {{- end }}
```

Reply:
as for now will merge it with the current version, to be consistent with other our charts, but for the future this looks like a good improvement todo 👍